File: Material Design/SupportDesignDemo/app/src/main/java/edu/cs4730/supportdesigndemo/SB_FABFragment.java
package edu.cs4730.supportdesigndemo;

import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.fragment.app.Fragment;

import com.google.android.material.snackbar.Snackbar;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;


/**
 * This example uses both a Floating Action button and a snackbar.
 * The fab is at the bttom right and it will move up and out of the way, so the snackbar will not
 * cover it up.  There is very little code in here to do it.  It's in the xml layout, android.support.design.widget.CoordinatorLayout
 * to do the work.  The snackbar is sent the coordinatorlayout, and the layout then "moves" the fab up (out of the way of snackbar)
 * so that there is no over lap.
 */
public class SB_FABFragment extends Fragment {

    Button btn;
    CoordinatorLayout mCoordinatorLayout;

    View.OnClickListener SBonClickListener = new View.OnClickListener() {
        public void onClick(View v) {
            Toast.makeText(requireContext(), "Yea!", Toast.LENGTH_SHORT).show();
        }
    };

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View myView = inflater.inflate(R.layout.fragment_sbfab, container, false);
        mCoordinatorLayout = myView.findViewById(R.id.coordinatorlayout1);
        btn = myView.findViewById(R.id.button02);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Snackbar.make(mCoordinatorLayout, "Did the FAB move?", Snackbar.LENGTH_LONG)
                    .setAction("Yes?", SBonClickListener)  //this line is optional!
                    .show();
            }
        });
        myView.findViewById(R.id.fab).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Snackbar.make(mCoordinatorLayout, "I knew you had to click the FAB.", Snackbar.LENGTH_LONG)
                    .show();
            }
        });

        return myView;
    }


}

File: Material Design/SupportDesignDemo_kt/app/src/main/java/edu/cs4730/supportdesigndemo_kt/SB_FABFragment.kt
package edu.cs4730.supportdesigndemo_kt

import androidx.coordinatorlayout.widget.CoordinatorLayout
import android.widget.Toast
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.fragment.app.Fragment
import com.google.android.material.snackbar.Snackbar
import edu.cs4730.supportdesigndemo_kt.databinding.FragmentSbfabBinding

/**
 * This example uses both a Floating Action button and a snackbar.
 * The fab is at the bttom right and it will move up and out of the way, so the snackbar will not
 * cover it up.  There is very little code in here to do it.  It's in the xml layout, android.support.design.widget.CoordinatorLayout
 * to do the work.  The snackbar is sent the coordinatorlayout, and the layout then "moves" the fab up (out of the way of snackbar)
 * so that there is no over lap.
 */
class SB_FABFragment : Fragment() {
    lateinit var binding: FragmentSbfabBinding

    var SBonClickListener =
        View.OnClickListener { Toast.makeText(requireContext(), "Yea!", Toast.LENGTH_SHORT).show() }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        binding = FragmentSbfabBinding.inflate(inflater, container, false)

        binding.button02.setOnClickListener {
            Snackbar.make(binding.coordinatorlayout1, "Did the FAB move?", Snackbar.LENGTH_LONG)
                .setAction("Yes?", SBonClickListener) //this line is optional!
                .show()
        }
        binding.fab.setOnClickListener {
            Snackbar.make(
                binding.coordinatorlayout1, "I knew you had to click the FAB.", Snackbar.LENGTH_LONG
            ).show()
        }
        return binding.root
    }
}